wincher-ab/wordpress-seo
# Feature/wincher upgrade callouts

### Testing steps:
- Apply the following patch to the [api](https://gitlab.int.wincher.com/wincher/solution/-/merge_requests/602) (to hit local /yoast/upgrade-campaign, not merged to prod yet). The current patch simulates a paying user. To simulate a free user you can remove changes made to the Yoast controller. 
[api.patch](https://github.com/Yoast/wordpress-seo/files/11206094/api.patch)
- Apply the following patch to the yoast branch (to use the local campaign endpoint).  [yoast.patch](https://github.com/Yoast/wordpress-seo/files/11206100/yoast.patch)
- We need to test a paying user + tracking >= 80% of keywords: no discount mentioned, only upgrade link, and we display a callout with the title
- We need to test a free user  + tracking >= 80% of keywords: callout + discount section to be displayed
- We need to test a user  + tracking < 80% of keywords: no callout.
- Callouts should be displayed in the SEO performance report in the dashboard, and the edit post page (both in page and modal)

### Note
- Based on https://app.shortcut.com/wincher/story/6212/yoast-dashboard-connect-button (PR: https://github.com/Yoast/wordpress-seo/pull/20107)
- Yoast repo PR (DRAFT): https://github.com/Yoast/wordpress-seo/pull/20144
- Yoast gonna need to use https://app.wincher.com/websites?plans=true&utm_medium=plugin&utm_source=yoast&referer=yoast&partner=yoast for https://yoa.st/wincher-upgrade

### Evidence
- paid user + 80% tracking

![image](https://user-images.githubusercontent.com/129404933/231321069-36a0623f-bfca-4080-b00f-2d718d439728.png)

- free account + 80% tracking / modal

![image](https://user-images.githubusercontent.com/129404933/231321124-905b9b6b-8cc5-445c-b815-e4cff7a6b170.png)

- free account + 80% tracking / edit page

![image](https://user-images.githubusercontent.com/129404933/231321224-371a3759-3247-4d6c-9803-ee5422b30c2a.png)

- free account + 80% tracking / dashboard page

![image](https://user-images.githubusercontent.com/129404933/231321271-12b4bacc-e218-4ac4-8374-c014f40483e6.png)
